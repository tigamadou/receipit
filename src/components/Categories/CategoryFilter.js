import React from 'react';
import PropTypes from 'prop-types';

const CategoryFilter = ({ filter, setFilter }) => (
  <div>
    <p>Filtrer by</p>
    <select onChange={(e) => setFilter(e.target.value)} value={filter}>
      <option>ASC</option>
      <option>DESC</option>
    </select>
  </div>
);

CategoryFilter.defaultProps = {
  setFilter: null,
  filter: 'ASC',
};

CategoryFilter.propTypes = {
  setFilter: PropTypes.func,
  filter: PropTypes.string,
};

export default CategoryFilter;
