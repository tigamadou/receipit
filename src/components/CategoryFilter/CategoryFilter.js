import React from 'react';
import PropTypes from 'prop-types';

const CategoryFilter = ({ filter, setFilter }) => (
  <div>
    <input onChange={(e) => setFilter(e.target.value)} value={filter} placeholder="Type your search here" />
  </div>
);

CategoryFilter.defaultProps = {
  setFilter: null,
  filter: '',
};

CategoryFilter.propTypes = {
  setFilter: PropTypes.func,
  filter: PropTypes.string,
};

export default CategoryFilter;
