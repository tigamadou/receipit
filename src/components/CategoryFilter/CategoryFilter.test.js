import React from 'react';
import { render, screen } from '@testing-library/react';
import CategoryFilter from './CategoryFilter';

describe('CategoryFilter component tests', () => {
  test('renders CategoryFilter component ', () => {
    render(<CategoryFilter />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('Filtrer by')).toBeInTheDocument;
  });
});
