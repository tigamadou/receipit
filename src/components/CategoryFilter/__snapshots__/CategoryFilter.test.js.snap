// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter component tests should match with snapshot 1`] = `
<div>
  <input
    className="filter"
    onChange={[Function]}
    placeholder="Type the recipe name"
    value=""
  />
</div>
`;
